### PR TITLE
fix: Availability Troubleshooter Only Pulls Default Availability (#13399)

### LINKED ISSUE #13399: [CAL-3017] Availability Troubleshooter Only Pulls Default Availability
If you go to a specific availability schedule and launch troubleshooter, it only pulls the availability from your default availability setting

Not sure if this matters as much as I don't think we even have the functionality to show the availability by the day — it's more of when you're free and busy but something to think about 

[https://www.loom.com/share/4367867331224a8eb5cf257824615fd6](https://www.loom.com/share/4367867331224a8eb5cf257824615fd6)

<sub>From [SyncLinear.com](https://synclinear.com) | [CAL-3017](https://linear.app/calcom/issue/CAL-3017/availability-troubleshooter-only-pulls-default-availability)</sub>

### PR DESCRIPTION
**Pull request by CrocodileZS**
## What does this PR do?
When launch troubleshooter from a specific availability schedule, it will pull the availability from the specific schedule rather than the default.

Fixes #13399 https://github.com/calcom/cal.com/issues/13399
https://www.loom.com/share/2bc82aee485a489c9805cf6ae0f0a4de?sid=36df0894-ae67-4551-8fd1-a88110239820

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## How should this be tested?

You could see my test video in loom video above.

## Mandatory Tasks

- [x] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.

### CODE REVIEW

File: packages/features/troubleshooter/components/EventScheduleItem.tsx [hunk 1]
```diff
@@ -4,19 +4,18 @@ import { useLocale } from "@calcom/lib/hooks/useLocale";
...
-import { useTroubleshooterStore } from "../store";
 import { TroubleshooterListItemHeader } from "./TroubleshooterListItemContainer";
 
 export function EventScheduleItem() {
   const { t } = useLocale();
-  const selectedEventType = useTroubleshooterStore((state) => state.event);
-
-  const { data: schedule } = trpc.viewer.availability.schedule.getScheduleByEventSlug.useQuery(
+  const scheduleIdStr = new URLSearchParams(window.location.search).get("scheduleId");
+  const scheduleId = scheduleIdStr ? Number(scheduleIdStr) : null;
```
Comment (Udit-takkar) [inline packages/features/troubleshooter/components/EventScheduleItem.tsx:12]: can we use useRouter here?

Comment (CrocodileZS) [reply packages/features/troubleshooter/components/EventScheduleItem.tsx]: Sorry for the super late reply. I'm a beginner for web development and will keep working on this issue. 
I have tried useRouterQuery here to get the scheduleId.

**Review by sean-brydon — CHANGES_REQUESTED** [review]
CleanShot 2024-05-07 at 09 48 10
Hi Loom looks great thanks for working on this, however, when i have tried testing this out myself with this new feature i can't get anything to load within the troubleshooter but i can when i checkout the latest main branch.

**Comment by keithwillcode** [comment]
Closing in favor of #14693

**Pull request closed without merging**